lib/ci-helper.ts

```diff
@@ -763,6 +772,211 @@ export class CIHelper {
+
+            if (commits) {
+                commits.map(node => {
+                    const login = node?.commit.committer?.user?.login ??
+                        node?.commit.author?.user?.login;
+
+                    // login may not be present if email unknown to github
+                    if (login && pr.author !== login) {
+                        throw new Error("Request author is not commit author.");
+                    }
+                });
+            }
+
+            const files = prInfo.files;
+            if (files && files.totalCount && files.nodes) {
+                files.nodes.map(node => {
+                    const changedFile = node as PullRequestChangedFile;
+                    this.checkNewUserFile(changedFile, files.totalCount,
+                        genLabel);
+                });
+            } else {
+                throw new Error("No files updated.");
+            }
+
+            const queryRef = `query {
+                repository(name: "${pr.baseRepo}", owner: "${pr.author}") {
+                    ref(qualifiedName: "${prInfo.baseRefName}") {
+                        name
+                    }
+                }
+            }`;
+
+            const refInfo = await this.github.graphql<queryResponse>(
+                pr.baseOwner,
+                queryRef
+            );
+
+            if (!refInfo || !refInfo.repository.ref) {
+                throw new Error(`Base branch ${prInfo.baseRefName
+                    } not found in ${pr.author}/${pr.baseRepo}.`);
+            }
+        } catch (e) {
+            if ((e as labelInfo).label === undefined ) {
+                const error = e as Error;
+                result = false;
+                await this.github.closePR(prInfo.url, `PR is being closed:\n\n${
+                    error.message}`);
+            } else {
+                const label = e as labelInfo;
+                await this.github.addPRLabels(prInfo.url, [label.label]);
+                await this.github.addPRComment(prInfo.url, `Label \`${
+                    label.label}\` added for this reason:\n\n${
+                    label.message}`);
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Perform some validation tests on a file in a pull request.
+     *
+     * @param changedFile file that is part of a commit in a pull request
+     * @param files number of files that are changed in the pull request
+     */
+    public checkNewUserFile(changedFile: PullRequestChangedFile,
+        files: number, setLabel: setLabelInfo):
+        void {
+        const filenameParts = path.parse(changedFile.path);
+
+        // one file with only deletions?
+        if (files === 1 &&
+            !changedFile.additions && changedFile.deletions) {
+            throw setLabel("validate", `File \`${changedFile.path
+                }\` expecting more than one change.`)
+        }
+
+        if (filenameParts.name.match(/.+\./)) {
```

Comment:
Wouldn't that catch `git.c`? It contains a dot that is not the first character.

Reply:
This is just the name (ie git) part of the base.

Reply:
Then I'm puzzled what this is supposed to do (and why the `+` is needed: `/.\./` should be idempotent).